### PR TITLE
Fix multi-depth picking for GeoJsonLayer

### PR DESCRIPTION
**Pull request by Pessimistress**
For #5295

#### Change List
- Fix clearPickingColor in PathLayer and SolidPolygonLayer when used as sub layers of a GeoJsonLayer
- Test

### CODE REVIEW

**Review by alasarr — APPROVED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -4205,6 +4212,16 @@ cyclist@^1.0.1:
   resolved "https://registry.yarnpkg.com/cyclist/-/cyclist-1.0.1.tgz#596e9698fd0c80e12038c2b82d6eb1b35b6224d9"
   integrity sha1-WW6WmP0MgOEgOMK4LW6xs1tiJNk=
 
+d3-array@^2.3.0:
+  version "2.9.1"
+  resolved "https://registry.yarnpkg.com/d3-array/-/d3-array-2.9.1.tgz#f355cc72b46c8649b3f9212029e2d681cb5b9643"
+  integrity sha512-Ob7RdOtkqsjx1NWyQHMFLtCSk6/aKTxDdC4ZIolX+O+mDD2RzrsYgAyc0WGAlfYFVELLSilS7w8BtE3PKM8bHg==
+
+"d3-color@1 - 2":
```
Comment (alasarr) [inline yarn.lock:4220]: I think these dependencies come from the Style Helpers commits at CARTO module (https://github.com/visgl/deck.gl/commit/7d6fa56b727956574eba87c9129756d34874f780). We could separate them in another commit, but since it's already in master I just keep them in this PR.

File: modules/layers/src/path-layer/path-layer.js [hunk 2]
```diff
@@ -190,6 +190,21 @@ export default class PathLayer extends Layer {
     return info;
   }
 
+  clearPickingColor(objectIndex) {
```
Comment (ibgreen) [inline modules/layers/src/path-layer/path-layer.js:193]: Just a general nit: Overriding methods and calling `super` is in my opinion a quite subtle design mechanism and I find hard to follow when reading a complex code base (which picking certainly is).

 It is just hard to see in the calling code that the flow may involve code in a subclass as there is no hint in the base method, no trail to follow.

It may be too much work but perhaps separating out this into two functions could remove the `super` call. One that does the basic clearing and one that is overridable and calls the basic clearing function, rather than `super`

File: modules/layers/src/solid-polygon-layer/solid-polygon-layer.js [hunk 3]
```diff
@@ -207,6 +207,21 @@ export default class SolidPolygonLayer extends Layer {
     return info;
   }
 
+  clearPickingColor(objectIndex) {
+    const {data} = this.props;
+
+    if (data[0] && data[0].__source) {
```
Comment (ibgreen) [inline modules/layers/src/solid-polygon-layer/solid-polygon-layer.js:213]: A helpful comment here maybe so that I don't have to think?

File: modules/layers/src/path-layer/path-layer.js [hunk 4]
```diff
@@ -190,6 +190,21 @@ export default class PathLayer extends Layer {
...
+    const {data} = this.props;
+
+    if (data[0] && data[0].__source) {
```
Comment (ibgreen) [inline modules/layers/src/path-layer/path-layer.js:196]: A helpful comment here maybe so that I don't have to think?

File: modules/core/src/lib/layer.js [hunk 5]
```diff
@@ -526,14 +526,13 @@ export default class Layer extends Component {
     model.setAttributes(shaderAttributes);
   }
 
-  // Sets the specified instanced picking color to null picking color. Used for multi picking.
-  clearPickingColor(color) {
+  // Sets the picking color at the specified index to null picking color. Used for multi picking.
+  clearPickingColor(objectIndex) {
```
Comment (ibgreen) [inline modules/core/src/lib/layer.js:530]: Reading this, I wonder if this function could be named better to improve readability. Something like `disablePickingIndex()`?

**Review by ibgreen — APPROVED** [review]
